File: Go/turkey_holiday_validator.go
package holidaychecker

// TurkeyHolidayValidator provides methods to check if a given date is a holiday in Turkey
type TurkeyHolidayValidator struct {
  *HolidayValidator
}

// NewTurkeyHolidayValidator creates a new TurkeyHolidayValidator with the option to include Saturdays as holidays
func NewTurkeyHolidayValidator(includeSaturday bool) *TurkeyHolidayValidator {
  return &TurkeyHolidayValidator{NewHolidayValidator(includeSaturday)}
}

File: Rust/turkey_holiday_validator.rs
use crate::holiday_validator::HolidayValidator;

pub struct TurkeyHolidayValidator {
    base_validator: HolidayValidator,
}

impl TurkeyHolidayValidator {
    pub fn new(include_saturday: bool) -> Self {
        Self {
            base_validator: HolidayValidator::new(include_saturday),
        }
    }

    pub fn is_holiday(&self, date: chrono::NaiveDate) -> bool {
        self.base_validator.is_holiday(date)
    }
}
